tests/pkcs11.py

```diff
@@ -0,0 +1,68 @@
+# Fake pkcs11 classes for simulation
+from PyKCS11 import PyKCS11Error
+
+from .settings import (VALID_KEY_ID, VALID_MECH, VALID_PIN, WRONG_KEY_ID,
+                       WRONG_MECH, WRONG_PIN)
+
+
+def _is_valid_mechanism(mechanism):
+  return mechanism._mech.mechanism == VALID_MECH._mech.mechanism and \
+      mechanism._param.hashAlg == VALID_MECH._param.hashAlg and \
+      mechanism._param.mgf == VALID_MECH._param.mgf and \
+      mechanism._param.sLen == VALID_MECH._param.sLen
+
+
+class _Session:
```

Comment:
Could you add some docstrings?

Reply:
I have added a link to `PyKCS11` session implementation.